File: admin_data/src/library/plus/pay.js
import Toast from '@/library/ui/ui-toast'
import Post from '@/library/ui/request-post'

/* 支付 */
export default (pay_type,url,data,callback,fail)=>{
  try{
    // APP支付
    if(pay_type=='alipay') data['type']='app';
    else if(pay_type=='wxpay') data['type']='APP';
    // 支付频道
    plus.payment.getChannels((channels)=>{
      let channel = null;
      for(let i in channels){
        if(channels[i].id==pay_type) channel=channels[i];
      }
      // 支付参数
      Post(url,data,(res)=>{
        const d = res.data;
        if(d.code!=0) return Toast(d.msg);
        // 唤起支付
        plus.payment.request(channel,d.data,callback,fail);
      });
    },(e)=>{
      return Toast('支付通道:'+e.message);
    });
  }catch(e){
    return Toast('H5方式:'+pay_type);
  }
}

File: flutter/lib/library/plus/pay.dart
import 'package:webmis/env.dart';
import 'package:webmis/library/ui/request.dart';
import 'package:tobias/tobias.dart';
import 'package:fluwx/fluwx.dart' as fluwx;

/* 支付类 */
class Pay{

  /* 微信支付 */
  static Future wxapy(String url, Map<String,dynamic> data) async {
    // 配置
    await fluwx.registerWxApi(
      appId: Env.pay['wx_appId'],
      doOnAndroid: true,
      doOnIOS: true,
      universalLink: Env.pay['universalLink'],
    );
    // 检测
    var isPay = await fluwx.isWeChatInstalled;
    if(isPay==false) throw '没有安装微信!';
    // 请求
    data['type']='APP';
    Map res = await post(url, data);
    if(res['code']!=0) throw res['code'];
    // 支付
    await fluwx.payWithWeChat(
      appId: res['data']['appid'],
      partnerId: res['data']['partnerid'],
      prepayId: res['data']['prepayid'],
      packageValue: res['data']['package'],
      nonceStr: res['data']['noncestr'],
      timeStamp: int.parse(res['data']['timestamp']),
      sign: res['data']['sign'],
    );
    // 回调
    fluwx.weChatResponseEventHandler.listen((res) {
      if (res is fluwx.WeChatPaymentResponse) {
        return res;
      }
    });
  }

  /* 支付宝 */
  static Future alipay(String url, Map<String,dynamic> data) async {
    // 检测
    var isPay = await isAliPayInstalled();
    if(isPay==false) throw '没有安装支付宝!';
    // 请求
    data['type']='app';
    Map res = await post(url, data);
    if(res['code']!=0) throw res['code'];
    // 支付
    Map<dynamic, dynamic> pay = await aliPayAuth(res['data']);
    return pay;
  }

}